app/controllers/health_cards_controller.rb

```diff
@@ -25,18 +25,30 @@ def qr_contents
 
   private
 
+  def jws
```

Comment:
All of this business logic should live somewhere outside of the controller.

Reply:
I agree with that and it pained me to put this in here, but we need access to the url helpers to create the fullUrls for the bundle and I can't think of a better place to do it. I used to do that in CovidHealthCard because it lived in Railsland, but now it's in the library. I didn't want to force the library to do a bunch of weird url manipulation because of rails

Reply:
I'm fine with it living in rails, I just don't think it should live in a controller.

Reply:
OK, now it is done in patient